kansha/card_addons/gallery/comp.py

```diff
@@ -77,15 +78,13 @@ def delete(self):
         DataAsset.remove_all(self.card.data)
         self.assets = []
 
-    def copy(self, parent, additional_data):
-        new_extension = self.__class__(parent, parent.action_log, self.assets_manager)
-        for data_asset in DataAsset.get_all(self.card.data):
-            new_asset_id = self.assets_manager.copy(data_asset.filename)
-            new_asset_data = DataAsset.add(new_asset_id, parent.data, additional_data['author'].get_user_data())
-            new_extension.create_asset(new_asset_data)
-            if data_asset.cover:
-                new_asset_data.cover = parent.data
-        return new_extension
+    def update(self, other):
+        for asset_comp in other.assets:
+            asset = asset_comp()
+            new_asset = self._add_asset(asset.file_info)
+            if asset.is_cover:
+                cropper = asset.cropper_info
+                self.make_cover(new_asset, **cropper)
```

Comment:
What happens in the case a cover was created before its position was stored and then the user copies the board?